### DISCUSSION TITLE
Actividad 3 - Grupo 2

### DISCUSSION THREAD

**Discussion by BenjaminPeres596** [frlp-utn-ingsoft/catedra#139] [Show and tell]
https://github.com/JuaniLezcano/vetsoft

**Comment by ivanduva** [comment]
Buenas!
La app anda bien y los tests pasan, pero creemos que hay algunos puntos a mejorar para que la actividad queda aprobada:

- En el caso de la address del Provider, no hay tests de integración, unidad ni e2e específicos para testear ese atributo. Los tests están bien pero son más generales sobre la clase Provider. Cuando se soluciona un issue, se debería agregar tests relativos a la funcionalidad que describía ese issue.
- La fecha de nacimiento de la Mascota queda vacía al editar, debería quedar en el form la que está guardada y no permitir que el campo quede vacío, más allá de que luego en la base de datos quede la anterior.
- El caso de la dosis de la medicina es igual a la de address del Provider, pero solo faltan los tests de unidad y e2e sobre ese atributo en específico. De unidad hay uno.

Cada problema debe solucionarlo la persona que se encargó de la funcionalidad.

Después hay algunos problemas relativos a la review que deberían mejorar tanto en las correcciones como en la próxima actividad porque son claves en cuanto a los conceptos que vimos en las clases:

- Hay muy pocos comentarios con relevancia. Como dijimos más de una vez, comentarios del estilo "Funciona bien" no aportan a la review. Para eso se pone el approve y listo.
- En este PR: https://github.com/JuaniLezcano/vetsoft/pull/4 alguien le hizo un fix al compañero que estaba trabajando sobre ese PR. El proceso más apropiado es justamente hacer una review, comentar el problema y que lo solucione el que está trabajando sobre eso. Además también vimos que otra persona hizo el merge. No le esquiven a la resolución de conflictos, si pasa esto de nuevo, vamos a mandar a hacer alguna corrección.

Les dejamos otros comentarios menores para que también los trabajen en las próximas entregas:

- Este commit: https://github.com/JuaniLezcano/vetsoft/pull/7/commits/8b0ae9d528c76bb5a4799e4cb9ba71ed1c81052d tiene un mal mensaje. El mensaje de commit debería ser una síntesis de la funcionalidad que ese commit resuelve.
- Hay Prs que tienen título y otros que solo contienen el nombre del branch. Estaría bueno unificar eso y que todos tengan un título apropiado sobre la funcionalidad.
- Les quedaron algunos prints que aparecen al correr la app
- En los tests e2e, ojo con repetir muchos expect que ya se chequearon en otros tests

Después valoramos varios puntos altos, como la estandarización de mensajes de commit y la cantidad de tests que se hicieron en general. También quedó muy buena la funcionalidad del stock.

Arreglando los comentarios del principio ya quedaría aprobada la actividad.
Buen trabajo!
Saludos

**Reply by BenjaminPeres596** [reply]
Buenas! Ya hicimos y subimos las correcciones.

**Comment by ivanduva** [comment]
Bien, queda aprobado entonces. 
Solo un comentario: el test de unidad sobre la dosis de la medicina podría probar que la función retorne el error además de fijarse si se creó o no la medicina.

Saludos!
